### ISSUE TITLE
reconnect delay  doesn't work when i disconnect network card

### ISSUE THREAD

**Issue by guillaume-fortin-keyneosoft** [blackhorse-one/stomp_dart_client#23]
If reconnect delay is not equals to 0, the stomp_handler must launch new connect.
this functionnality work when a TimeoutException is occured but not when a WebSocketChannelException is occured.

In my case, i disconnect my network adapter, a WebSocketChannelException occure, and the stomp handler doesn't start a new connect after reconnect delay.

Can you change the code from StompHandler start method like that:

```
on WebSocketChannelException catch (err) {      
      if (config.reconnectDelay == 0) {
        _onError(err);
      } else {
        config.onDebugMessage('Connection web socket error...reconnecting');
        _onDone();
      }
    }
```

full code
```
void start() async {
    try {
      channel = await platform.connect(config);
      channel.stream.listen(_onData, onError: _onError, onDone: _onDone);
      _connectToStomp();
    } on WebSocketChannelException catch (err) {      
      if (config.reconnectDelay == 0) {
        _onError(err);
      } else {
        config.onDebugMessage('Connection web socket error...reconnecting');
        _onDone();
      }
    } on TimeoutException catch (err) {
      if (config.reconnectDelay == 0) {
        _onError(err);
      } else {
        config.onDebugMessage('Connection timed out...reconnecting');
        _onDone();
      }
    } catch (err) {
      _onDone();
    }
  }
```

Sorry, i can't pull request because i am currently adding capability to connect to STOMP with SOCKJS.
I will pull request SOCKJS parser in few days.
you can watch the changes [here](https://github.com/keyneosoft/stomp_dart_client/tree/Feature/add_sockjs_support)

**Comment by guillaume-fortin-keyneosoft** [comment]
Done to Pull request #24

**Comment by KammererTob** [comment]
Merged your PR and released 0.3.6
